cdc/generation.cc

```diff
@@ -330,7 +330,7 @@ topology_description limit_number_of_streams_if_needed(topology_description&& de
     return topology_description(std::move(entries));
 }
 
-future<cdc::generation_id> make_new_cdc_generation(
```

Comment:
rename generation_id -> generation_id_v1 should be a separate commit!!!

Reply:
That's not logically a rename, some places still use `generation_id` (which is now a variant), this place stops using `generation_id` and starts using `generation_id_v1`. The fact that `generation_id_v1` is the same type as `generation_id` was before this commit is a "coincidence".

If I did a commit that renames `generation_id` to `generation_id_v1` before this commit then in this commit I would have to rename some instances `generation_id_v1` back to `generation_id`. Hard and unnecessary work.